### ISSUE TITLE
[Announcement] Discontinuation of development for Gameyfin v1

### ISSUE THREAD

**Issue by grimsi** [gameyfin/gameyfin#234]
**TL;DR: Gameyfin v1 will only receive bug fixes and security patches from now on - Gameyfin v2 is in development, but will take time.**

Gameyfin was and is a tool inspired by [Grifter](https://github.com/terrybrash/grifter) to easily manage and distribute games for LAN parties.
It fulfills this purpose very well and for many users the functionality of the software is sufficient. In recent months, however, it has become clear that a large number of users would also like more from Gameyfin in order to make it a fully comprehensive management tool for their very own game library.
While implementing a new feature (dynamic configuration) I noticed that **due to design decisions made at the beginning of the development, the integration of additional features is much more difficult than it should be**: Gameyfin has been built around IGDB's API, so it is not possible to (easily) integrate other metadata sources. The choice of technologies used was strongly influenced by my knowledge and interests at that time and today I would choose differently in some aspects.

**This means that I will now discontinue development on Gameyfin v1.** However, this does not mean the complete end of Gameyfin: I have started working on Gameyfin v2, which will (hopefully) fix the shortcomings and integrate lessons learned from the development of version 1.
Gameyfin v2 will be a complete rewrite, especially the frontend will be more modern and beautiful. The admin interface will be completely expanded (currently it is just a kind of placeholder for functions that never came).

Gameyfin v2 will have the following **new additional features** on release (in addition to the existing functionality of Gameyfin v1):
* More modern UI with better accessibility features ✅
* More detailed documentation
* User management (incl. permission management) ✅
* SSO (most likely OICD) ✅
* Complete configuration in the UI instead of via environment variables ✅
* Support for various metadata sources based on plugins (which can also be developed by the community)
* Prometheus metrics

Some other features that are planned, but will probably not be part of the initial release of v2:
* Tracking of own metrics (played/not played, personal rating)
* Community features (e.g. recommend games to friends)
* Support for custom data (mods, save-games, ...)
* Support for additional download methods (SMB, local torrent)
* Better support for different platforms
* Better support for different versions of games in parallel

There are also some things that will (probably) never be integrated into Gameyfin:
* Client software (except plugins for clients like Playnite) and with that:
  * Tracking of playtime
  * Automatic installation
  * Achievements
* Download and import of games from the internet/usenet/torrents directly into Gameyfin (similiar to Radarr/Sonarr)

As you can see, several new features are planned, which will take time to develop. So please be patient, **I will continue to maintain the project exclusively in my spare time and therefore have limited time**. Bugs and security issues will be fixed, but no new features will be released.
Currently there is no public release date of any beta or final versions planned. It will be ready when it's ready.

Thanks again for your great support (without it development would have stopped two years ago)!

-  Simon (grimsi)

**Comment by grimsi** [comment]
Here is a sneak peek of Gameyfin v2.
Multi-user support (including role-based permissions) is already working 🙂 
Still a lot of work ahead, especially the SSP and plugin parts are very complex.

![grafik](https://github.com/grimsi/gameyfin/assets/9295182/ff44d425-4e91-41a6-a65d-f58e174fed5c)

**Comment by grimsi** [comment]
Since I finally found some time last week to continue my work on v2, I thought I would share an update with you.
I'm currently finished the fundamental functionality such as user management, permissions and just today I implemented SSO via OAuth2/OIDC 😄 

The frontend is looking very clean in my opinion and I am finally getting used to React.
Spring Boot is a godsend as always and Kotlin is the cherry on top of it.

So there is stil quite some work ahead and I can't guarantee that I can sustain the pace I had for the last days since I had a lot more time than usual on my hands.

Anyway, here are a few quick previews for those of you who don't want to run a local dev instance 😉 

Theming:
https://github.com/user-attachments/assets/c20a2194-f4aa-4c06-9973-5594f2eec88e

Login page (if not using SSO):
![Screenshot 2024-09-16 170625](https://github.com/user-attachments/assets/97209f6c-bb0e-4ce9-9901-233a1caba70e)

SSO: 
https://github.com/user-attachments/assets/df20ae4f-c288-48e5-8135-1f2952342205

![Screenshot 2024-09-16 164402](https://github.com/user-attachments/assets/bc3e7d37-d2bb-47fd-b0fe-404aa598adde)

User administration:

![Screenshot 2024-09-16 165626](https://github.com/user-attachments/assets/69e365bb-9f62-4d8f-beab-b41562a34d13)

**Comment by grimsi** [comment]
Another milestone after I took a longer pause from coding (I got a kitten): The IGDB plugin is now functionally complete (as in it has the same functionality as the IGDB integration in Gameyfin v1) 🎉 

The new plugin system allows you to implement your very own metadata provider (and maybe more stuff in the future) or install them from other users (a "plugin store" is planned, but not vor 2.0).

Also I made a (very basic) implementation for a plugin fetching metadata using Steams (undocumented) public API. It works very well, but since the API is not documented I have to reverse engineer it a bit more to get all the information I need. On the plus side it will not need any credentials and has no rate limits as far as I can tell.

BTW on this screenshot you can also see the new "colorblind" theme 😎 

![{0A617FD2-45D9-4E71-BB8F-79AF69A5F2EA}](https://github.com/user-attachments/assets/0fd4e315-7e42-4600-80eb-d9336448f8a7)

**Comment by grimsi** [comment]
About a year ago I decided that rewriting Gameyfin from scratch is the best way to improve it.
To be honest I didn't think it would take this much time since I wrote the original version of Gameyfin in about a month or so.
However I learned a few lessons now (there are reasons it didn't take so long for v1) and Gameyfin v2 offers much more functionality. To be honest you could almost think it's a completely new app.
I switched from Java to Kotlin in the backend and from Angular to React in the frontend while having worked with neither of both before, just for the sake of learning. Maybe not the best solution for a quick result, but I really had fun working with the new tech stack, especially [Vaadin Hilla](https://vaadin.com/hilla) was a godsent.
So why am I writing this long text? I don't want to overpromise again, but I finally feel like development is on the home stretch. Give or take a few weeks, but Gameyfin v2 will come.

Here are some fresh screenshots to show what I have been cooking:

Homepage (WIP):
![Image](https://github.com/user-attachments/assets/99445fbc-61ad-4da9-8821-bd3c6743e7ae)

Single game view (very much WIP):
![Image](https://github.com/user-attachments/assets/1554d129-4e54-4d94-9927-721d42688a58)

Theming (including colorblind support):
![Image](https://github.com/user-attachments/assets/ce218ba4-fe45-4b95-bd10-f645a6788bac)

Admin UI for libraries (not all options shown are already implemented):
![Image](https://github.com/user-attachments/assets/a656c744-2068-454d-9bc1-382f6c460fa6)

Admin UI for SSO:
![Image](https://github.com/user-attachments/assets/4144ddfb-d435-4423-8c06-83e85d3d6ae5)

Admin UI for plugins (including all plugins I developed until now):
![Image](https://github.com/user-attachments/assets/4e6f45fb-d660-4a81-b62f-55b22aa7924b)


I know that while Gameyfin was more or less the first of its kind there are now multiple similiar apps, some ready, some still in development.
With Gameyfin I would like to focus on one thing: **Turn your game libraries into a web page.**
Thats it. Maybe there will be some minor social features, but it certainly wont be the focus.


Feel free to tell me what you think about it!

**Comment by grimsi** [comment]
Preview thread for Gameyfin v2: https://www.reddit.com/r/gameyfin/comments/1kkb001/preview_gameyfin_v2/
